#!/usr/bin/env python3
PACKAGE = "asv_sim"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("max_rpm",     double_t, 0, "Maximum RPM",                            3200.0, 0.0, 20000.0)
gen.add("max_power",   double_t, 0, "Max power, in Watts",                    8948.4, 0.0, 1000000000.0)
gen.add("idle_rpm",    double_t, 0, "Idle RPM",                                  0.0, 0.0, 5000.0)
gen.add("max_rpm_change_rate",
                                double_t, 0, "Maximum RPM change rate, in RPM/sec",    1000.0, 1.0, 100000.0)
gen.add("max_speed",   double_t, 0, "Maximum speed, in m/s",                    2.75, 0.0, 50.0)
gen.add("mass",        double_t, 0, "Mass, in kg",                            2000.0, 1.0, 90000000.0)
gen.add("max_rudder_angle",
                                double_t, 0, "Maximum rudder deflexion, in degrees",     30.0, 0.0, 90.0)
gen.add("rudder_distance",
                                double_t, 0, "Distance between rudder and center of mass, in meters",
                                                                                          2.0, 0.0, 200.0)
gen.add("rudder_coefficient",
                                double_t, 0, "Rudder effectiveness",                     0.25, 0.0, 1.0)

exit(gen.generate(PACKAGE, "asv_sim", "dynamics"))
